### ISSUE TITLE
통합 테스트 

### ISSUE THREAD

**Issue by robertchoi80** [openinfradev/promote#1]
HANU 플랫폼의 Release 작업을 위해, 또는 주기적인 검증을 위해 통합 테스트 수행이 필요한데, 이에 대한 논의가 필요함

통합테스트 시에는 다음의 컴포넌트들이 모두 검증되어야 함. 이를 각각 별도의 job에서 검증 가능하도록 하되, 필요할 경우 meta job으로 묶어 일괄 수행할 수 있도록 함

- Kubernetes: sonobuoy 테스트 + custom plugin?
-- sonobuoy에서 기본적으로 제공하는 test set을 확인하고, 부족할 경우 custom plugin 추가 필요
- Apps
-- lma: ??
-- openstack : (기존 job 활용 가능) Rally 시나리오 테스트 + Resiliency test (/w cookie-monster?) + (optional) API test (/w tempest)

**Comment by Jaesang** [comment]
`admin-tools`는 이 논의에서 제외되나요?

**Comment by robertchoi80** [comment]
아닙니다. 자유롭게 추가하셔도 됩니다^^

**Comment by intelliguy** [comment]
LMA의 경우는 예전에 openstack-helm 쪽에서 사용했던 것들부터 시작할수 있을듯 합니다.
기억에 의존하는 것이므로 정확하지는 않으나 정리해 보면 다음과 같습니다.

### EFK의 경우
* Elasticsearch 의 테스트는 helthcheck rest를 호출하는 것
* Fluent의 경우는 임의의 로그를 발생시키고 이를 fluent -> es로 잘 전달되는 것을 확인하는 것

그외에는 따로 없었던거 같긴한데 한번 찾아보는 것도 좋을듯 합니다.

일단 일반화 해보면
* 소프트웨어 설치에 대한 검증
* pipeline에 대한 검증
* ....

등으로 분할해서 소프트웨어별 검증 방법을 정리해야 할 듯 합니다.

**Comment by robertchoi80** [comment]
네~ 말씀하신 소프트웨어별 검증 방법을 논하는게 목적입니다. 일단 큰 틀이 잡히면 task 분담해서 각각 맡은 사람이 좀더 deep하게 고민해서 설계한뒤 전체 공유하는 식으로 진행하면 좋을 거 같아요.

**Comment by Jaesang** [comment]
keycloak은 제공되는 API를 통해 CRUD 테스트가 가능해보입니다.
